buildserver/build_server.go

```diff
@@ -178,6 +178,19 @@ func (h *BuildHandler) Handle(ctx context.Context, conn *jsonrpc2.Conn, req *jso
 			return nil, err
 		}
 
+		// Fall back to reading the `originalRootURI` from
+		// `InitializeParams.intitializationOptions` if absent from the
+		// `InitializeParams`. This makes go-langserver more LSP-compliant so that
+		// clients don't need to go outside of the LSP specification to communicate
+		// with it.
```

Comment:
Note that this is still not really compliant from a semantics perspective because `initializationOptions` is intended to be options you would pass through CLI arguments, i.e. not values that change for every workspace.